app/code/core/Mage/Admin/Model/User.php

```diff
@@ -395,7 +397,7 @@ public function authenticate($username, $password)
             $this->loadByUsername($username);
             $sensitive = ($config) ? $username == $this->getUsername() : true;
 
-            if ($sensitive && $this->getId() && Mage::helper('core')->validateHash($password, $this->getPassword())) {
+            if (Mage::helper('core')->validateHash($password, $this->getPassword()) && $sensitive && $this->getId()) {
```

Comment:
Testing sensitive first would be faster than testing validateHash

Reply:
That's why I swapped it. It can be possible to check if the username exists if its the other way around. AFAIK one couldn't tell with simple hashes (sha256/512) but bcrypt is quiet a bit slower.